lib/howl/commands/app_commands.moon

```diff
@@ -665,39 +665,107 @@ do_search = (search, whole_word) ->
       m = matches[i]
       file_search_hit_to_location(m, search, display_as)
 
-  locations, searcher, project
+  locations, searcher
 
-command.register
-  name: 'project-file-search',
-  description: 'Searches files in the the current project'
-  input: (opts) ->
-    editor = app.editor
-    search = nil
-    whole_word = false
+ask_for_search_directory = (search_query) ->
+  buffer = app.editor.buffer
+  file = buffer.file or buffer.directory
+  project = Project.for_file(file) if file
 
-    if opts.text and not opts.text.is_empty
-      search = opts.text
+  opts =
+    title: if search_query
+      "(Please specify a directory to search for '#{search_query}'): "
+    else
+      "(Please specify a directory to search): "
+    prompt: "Directory: "
 
-    unless search or app.window.command_panel.is_active
-      if editor.selection.empty
-        search = app.editor.current_context.word.text
-        whole_word = true unless search.is_empty
-      else
-        search = editor.selection.text
+  start_dir = if project
+    project.root
+  else
+    get_buffer_dir(buffer)
+
+  if start_dir
+    opts.path = start_dir.path
+
+  interact.select_directory opts
 
-    if not search or search.is_empty
-      search = interact.read_text prompt: opts.prompt
+get_search_query = (opts) ->
+  editor = app.editor
+
+  search = nil
+  whole_word = false
+
+  if opts.text and not opts.text.is_empty
+    search = opts.text
+
+  unless search or app.window.command_panel.is_active
+    if editor.selection.empty
+      search = editor.current_context.word.text
+      whole_word = true unless search.is_empty
+    else
+      search = editor.selection.text
+
+  if not search or search.is_empty
+    search = interact.read_text prompt: opts.prompt
 
-    if not search or search.is_empty
-      log.warn "No search query specified"
-      return
+  search, whole_word
 
-    locations, searcher, project = do_search search, whole_word
-    if #locations > 0
-      interact.select_location
-        title: "#{#locations} matches for '#{search}' in #{project.root.short_path} (using #{searcher.name} searcher)"
-        items: locations
+-- Common input handler for search commands displaying results
+-- in a pop-up panel. E.g. project-file-search and folder-search
+search_input = (opts) ->
+  search, whole_word = get_search_query opts
+
+  if not search or search.is_empty
+    log.warn "No search query specified"
+    return
+
+  directory = opts.directory or ask_for_search_directory search
+  if not directory
+    log.warn "No directory selected"
+    return
+
+  locations, searcher = do_search directory, search, whole_word
+
+  if #locations > 0
+    interact.select_location
+      title: "#{#locations} matches for '#{search}' in #{directory.short_path} (using #{searcher.name} searcher)"
+      items: locations
```

Comment:
Not sure how the original code behaves, but would be nice to have an `else` case here that just shows "No matches" or something like that.

Reply:
I haven't changed anything here compared to the original version. do_search takes care of the "No matches found" message.